pagegraph-cli/src/request_id_info.rs

```diff
@@ -68,17 +69,21 @@ pub fn main(graph: &PageGraph, request_id_arg: usize, frame_id: Option<FrameId>,
     let execute_edge = graph.outgoing_edges(start_source)
         .filter(|edge| matches!(edge.edge_type, EdgeType::Execute {})).nth(0);
     let script_node = execute_edge.map(|x| graph.target_node(x));
+    // If the node is not a script node, then fail silently
 
     let request_info = if let EdgeType::RequestComplete { resource_type, status, response_hash, headers, size, .. } = &complete_edge.edge_type {
         if let EdgeType::RequestStart { request_type, .. } = &start_edge.edge_type {
             if let NodeType::Resource { url } = &start_target.node_type {
-                let source = script_node.map(|script_node| {
-                    if let NodeType::Script { source, .. } = &script_node.node_type {
-                        source.clone()
-                    } else {
-                        unreachable!()
+                let source = match script_node {
+                    None => process::exit(1), // fail silently
```

Comment:
would a panic!() be better?

Reply:
The way I see it is:

- an ideally-designed program should not panic or otherwise subvert normal control flow (e.g. with `process::exit`)
- `panic!()`, `unwrap()`, etc. can be convenient to add as _temporary_ fill-ins for unhandled paths in the program
- `panic!()` has nicer ergonomics in Rust than `process::exit` (e.g. format string support, line numbers, etc.)

I think the "correct" thing to do here would be to take advantage of `Option`/`Result` types, but that can sometimes require more engineering than it's worth - in which case panics would be preferred.

Reply:
Agreed - I've raised #18 so that we can track this known issue with `request_id_info` (though the fix will probably be in pagegraph graph encoding).